### DISCUSSION TITLE
How do slash commands work if the bot is not in the channel?

### DISCUSSION THREAD

**Discussion by ghost** [DisnakeDev/disnake#1144] [Q&A]
I couldn't figure out how disnake works with slash or is discord weirding out here, but it's not clear to me how slash commands can work where the bot doesn't actually exist, is there any solution for this? I wouldn't want to be stuck with channel ids, because that's not a solution when there are 100 text channels on the server.

In the arguments for slash, I unfortunately found nothing clear to me. 

![Screenshot](https://github.com/DisnakeDev/disnake/assets/64988751/5edc1be1-22bf-4c8b-959e-6b9af8a7e90b)

**Comment by elenakrittik** [comment]
The "be in channel for commands to work" requirement only applies to prefix (text/"old-style") commands, as they scan incoming text messages to detect command invocations (and if the bot is not in a channel, discord doesn't send the bot any notifications about messages in that channel). Application commands (including slash ones) do not have this limitation as they are registered per-guild, and therefore are available guild-wide.

**Reply by ghost** [reply]
So the only solution is to write an ephemeral message to the user that the bot has no permissions for the channel (conditionally for viewing), thus making a plug (a feature / feature / not a bug, =0)?

**Reply by elenakrittik** [reply]
I haven't encountered a case like this until now, but from a quick test it seems that bot will be able to respond to the slash command just fine, even if it can't "see" the channel normally. Some information about the channel will be present too, but i don't think you'll be able to, e.g., fetch messages from it (as demonstrated by the absence of `inter.channel.last_message`). If you indeed need some info not available in case of running in a "secret" channel, then yes, your best option would be to tell the user that the bot doesn't have enough permissions to function.

![изображение](https://github.com/DisnakeDev/disnake/assets/77104725/3eb63ffa-3b8c-4667-a42f-0e0ba3a0cec7)
![изображение](https://github.com/DisnakeDev/disnake/assets/77104725/413c48dd-234e-438f-b613-4143b3e500ec)
![изображение](https://github.com/DisnakeDev/disnake/assets/77104725/0f6568e4-ac34-466d-9bcc-427b74efadd0)

**Reply by ghost** [reply]
Okay, thanks for some clarification. I'll leave the option that the bot will alert the user.
